---
title: dRNAmain
author: Xiaofeng Chen
date: 2024-5-8
category: Jekyll
layout: post
mermaid: true
---
dRNAmain moudle includes fast5-basecall, mapping and transcriptome count.
<h2>Concept</h2>


<h2>build of reference file</h2>
The ref in config.yml need to be build in manual, requiring fastq file and annotation file(in gtf/gff3)
Since the following dRNAas needs gtf file, we advise you to choose gtf annotation file and use gffread to convert to gff3.
```shell
gffread genome.gtf -o- > genome.gff3
```
Then generate fastq file for transcriptome mapping.

```shell
gffread -F -w transcriptome.fa -g genome.fa genome.gff3 
```
<h2>selection of f5c version</h2>
For f5c, it offers two version called f5c_x86_64_linux and f5c_x86_64_linux_cuda.
cuda version is for GPU, so if you only run with CPU, please change dRNAmain.py line 159

```shell
f5c_x86_64_linux_cuda index --slow5 {input[1]} {input[0]}
```
to
```shell
f5c_x86_64_linux index --slow5 {input[1]} {input[0]}
```

<h2>Choice of cfg files</h2>
The cfg files are located in /ont-guppy/data/
You can assign the specific file based on the chips' version your experinment choose.
